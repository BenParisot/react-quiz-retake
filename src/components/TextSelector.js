import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ handleHeaderChange, handleFooterChange, handleImageSrcChange }) {
  return (
    <form>
      <label>
        Header Text: <input onChange={handleHeaderChange}></input>
      </label>
      <label>
        Footer Text: <input onChange={handleFooterChange}></input>
      </label>
      <label>
        Image Src: <input onChange={handleImageSrcChange}></input>
      </label>
    </form>
  );
}
TextSelector.propTypes = {
  handleHeaderChange: PropTypes.func.isRequired,
  handleFooterChange: PropTypes.func.isRequired,
  handleImageSrcChange: PropTypes.func.isRequired
};

export default TextSelector;
