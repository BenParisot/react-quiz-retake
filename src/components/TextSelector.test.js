import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector component', () => {
  it('displays the textselector', () => {
    const handleHeaderChange = () => {return 'hi';};
    const handleFooterChange = () => {return 'hi';};
    const handleImageSrcChange = () => {return 'hi';};
    const wrapper = shallow(<TextSelector handleHeaderChange={handleHeaderChange} handleFooterChange={handleFooterChange} handleImageSrcChange={handleImageSrcChange} />);
    expect(wrapper).toMatchSnapshot(); 
  });
});
