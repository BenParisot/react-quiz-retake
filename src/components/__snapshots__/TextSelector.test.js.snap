// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component displays the textselector 1`] = `
<form>
  <label>
    Header Text: 
    <input
      onChange={[Function]}
    />
  </label>
  <label>
    Footer Text: 
    <input
      onChange={[Function]}
    />
  </label>
  <label>
    Image Src: 
    <input
      onChange={[Function]}
    />
  </label>
</form>
`;
